core/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+pub mod basic_config;
+pub mod configuration;
+pub mod projects;
+
+extern crate anyhow;
+extern crate serde;
+extern crate shellexpand;
+
+use anyhow::Context;
+use anyhow::Result;
+use anyhow::anyhow;
+use basic_config::BasicConfigContent;
+use std::fs;
+use std::path::Path;
+use std::vec::Vec;
+
+pub fn setup(path: &Path, additional_modules: Vec<Box<dyn SetupAdapter>>) -> Result<()> {
+    if path.is_dir() {
+        return Err(anyhow!(
+            "This command needs a single file, that is not a directory!"
+        ));
+    }
+
+    let config_content: String = fs::read_to_string(path)
+        .with_context(|| format!("Cannot read file at path {}", path.display()))?;
+
+    let Some(extension) = path.extension().and_then(|ext| ext.to_str()) else {
+        return Err(anyhow!("Could not read extension "));
+    };
+
+    let config = configuration::generate_config::<BasicConfigContent>(&config_content, extension)?;
+
+    projects::configure_project_base_path(&config.base.clone())?;
+
+    for module in additional_modules {
+        let _ = module
+            .exec(&config_content, extension, &config)
+            .with_context(|| "Something went terribly wrong");
```

Comment:
I don't think this will ever return your something went wrong error. I think if you want to continue in the face of errors, you should continue capturing the exec result into `let _` (this catches Err and does nothing). If you want to exit the function early you should instead do:

```suggestion
        module
            .exec(&config_content, extension, &config)
            .with_context(|| "Something went terribly wrong")?;
```

I think clippy will allow you to omit the `let _ =` if you have a try questionmark, but I'm not sure.